src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -96,12 +96,14 @@ internal interface IMessageSource
     internal abstract class HostEnvironmentBase<TEnv> : ChannelProviderBase, IHostEnvironment, IChannelProvider
         where TEnv : HostEnvironmentBase<TEnv>
     {
+        public bool IsCancelled { get; set; }
```

Comment:
I don't think this should be publicly settable.

Reply:
Its within an internal class so its internal right?

---
In reply to: [264861769](https://github.com/dotnet/machinelearning/pull/2797#discussion_r264861769) [](ancestors = 264861769)

Reply:
Correct. And we don't want internal code to be able to set this property. Only this class (or subclasses) should be able to set it.

Reply:
Yep, made it private set.

---
In reply to: [264863680](https://github.com/dotnet/machinelearning/pull/2797#discussion_r264863680) [](ancestors = 264863680)